packages/paypal-commerce-integration/src/PaypalCommerceRatePayPaymentMethod.tsx

```diff
@@ -0,0 +1,156 @@
+import React, {FunctionComponent, useCallback, useEffect, useMemo, useRef} from 'react';
+
+import {
+    PaymentMethodProps,
+    PaymentMethodResolveId,
+    toResolvableComponent,
+} from '@bigcommerce/checkout/payment-integration-api';
+import { DynamicFormField, DynamicFormFieldType, FormContext } from "@bigcommerce/checkout/ui";
+import { FormField } from '@bigcommerce/checkout-sdk';
+import getPaypalCommerceRatePayValidationSchema from './validation-schemas/getPaypalCommerceRatePayValidationSchema';
+
+interface RatePayFieldValues {
+    ratepayBirthDate: string;
+    ratepayPhoneCountryCode: string;
+    ratepayPhoneNumber: string;
+}
+
+const formFieldData: FormField[] = [
+    {
+        name: 'ratepayBirthDate',
+        custom: false,
+        id: 'ratepayBirthDate',
+        label: 'payment.ratepay.birth_date',
+        required: true,
+        fieldType: DynamicFormFieldType.DATE,
+    },
+    {
+        name: 'ratepayPhoneCountryCode',
+        custom: false,
+        id: 'ratepayPhoneCountryCode',
+        label: 'payment.ratepay.phone_country_code',
+        required: true,
+        fieldType: DynamicFormFieldType.TEXT,
+        type: 'string',
+        maxLength: 2,
+    },
+    {
+        name: 'ratepayPhoneNumber',
+        custom: false,
+        id: 'ratepayPhoneNumber',
+        label: 'payment.ratepay.phone_number',
+        required: true,
+        fieldType: DynamicFormFieldType.TEXT,
+        maxLength: 11,
+        min: 8,
+    }
+];
+
+const PaypalCommerceRatePayPaymentMethod: FunctionComponent<any> = ({
+    method,
+    checkoutService,
+    paymentForm,
+    onUnhandledError,
+    paymentForm: {
+        isSubmitted,
+        setFieldValue,
+        setValidationSchema,
+        setSubmitted,
```

Comment:
I saw that you have 2 different styles of using paymentForm methods:
- using methods without paymentForm like `setFieldValue`;
- using method with paymentForm, like `paymentForm.hidePaymentSubmitButton(method, true)`;

It might be better to use one code style to avoid mess in code. Please, choose the variant that you like and update the code below

Reply:
removed OnrenderButton method as it not needed